ui/pages/confirmation/confirmation.js

```diff
@@ -258,14 +260,25 @@ export default function ConfirmationPage({
     return INPUT_STATE_CONFIRMATIONS.includes(type);
   };
 
-  const handleSubmit = () =>
-    templateState[pendingConfirmation.id]?.useWarningModal
+  const handleSubmit = async () => {
+    setLoading(true);
+    const submitResult = await (templateState[pendingConfirmation.id]
+      ?.useWarningModal
       ? setShowWarningModal(true)
-      : templatedValues.onSubmit(
+      : await templatedValues.onSubmit(
           hasInputState(pendingConfirmation.type)
             ? inputStates[MESSAGE_TYPE.SNAP_DIALOG_PROMPT]
             : null,
-        );
+        ));
+
+    if (submitResult?.success) {
```

Comment:
It looks like this component will be stuck in the "loading" state if the submit result is missing this property. Yet only `add-ethereum-chain` returns this. The other confirmations would all get stuck.

Reply:
this is done this way to disable the button so user can only click 'Cancel'

Reply:
Right, but for the non-`add-ethereum-chain` confirmations there will be no way to proceed even if there is no error